nats-base-client/serviceclient.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2022 The NATS Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  Empty,
+  NatsConnection,
+  RequestManyOptions,
+  RequestStrategy,
+} from "./types.ts";
+import { QueuedIterator, QueuedIteratorImpl } from "./queued_iterator.ts";
+import {
+  JSONCodec,
+  SchemaInfo,
+  ServiceInfo,
+  ServiceStats,
+  ServiceVerb,
+} from "./mod.ts";
+import { ServiceIdentity, ServiceImpl, ServiceSchema } from "./service.ts";
+
+export interface ServiceClient {
+  /**
+   * Pings services
+   * @param name - optional
+   * @param id - optional
+   */
+  ping(name?: string, id?: string): Promise<QueuedIterator<ServiceIdentity>>;
+
+  /**
+   * Requests all the stats from services
+   * @param name
+   * @param id
```

Comment:
missing `version` according to https://github.com/nats-io/nats-architecture-and-design/blob/service-api/adr/ADR-33.md#standard-field

Reply:
the version is for the returned data, for the client only the name/id are options for discovery etc.